packages/medusa/src/api/routes/admin/products/get-product.ts

```diff
@@ -80,9 +92,28 @@ export default async (req, res) => {
 
   const product = rawProduct
 
+  const decoratePromises: Promise<any>[] = []
   if (!shouldSetPricing) {
-    await pricingService.setProductPrices([product])
+    decoratePromises.push(pricingService.setProductPrices([product]))
+  }
```

Comment:
comment: Are we not using the flag incorrectly? I would assume the exact opposite behavior i.e. `shouldSetPricing -> setProductPrices`

Reply:
Yes 😂 updated it